crates/bevy_ecs/src/observer/mod.rs

```diff
@@ -494,40 +494,72 @@ impl ObserverTrigger {
     }
 }
 
-// Map between an observer entity and its runner
-type ObserverMap = EntityHashMap<ObserverRunner>;
+/// Map between an observer entity and its [`ObserverRunner`]
+pub type ObserverMap = EntityHashMap<ObserverRunner>;
 
-/// Collection of [`ObserverRunner`] for [`Observer`] registered to a particular trigger targeted at a specific component.
+/// Collection of [`ObserverRunner`] for [`Observer`] registered to a particular event targeted at a specific component.
 ///
 /// This is stored inside of [`CachedObservers`].
 #[derive(Default, Debug)]
 pub struct CachedComponentObservers {
-    // Observers listening to triggers targeting this component
-    map: ObserverMap,
-    // Observers listening to triggers targeting this component on a specific entity
-    entity_map: EntityHashMap<ObserverMap>,
+    // Observers listening to events targeting this component, but not a specific entity
+    global_observers: ObserverMap,
+    // Observers listening to events targeting this component on a specific entity
+    entity_component_observers: EntityHashMap<ObserverMap>,
 }
 
-/// Collection of [`ObserverRunner`] for [`Observer`] registered to a particular trigger.
+impl CachedComponentObservers {
+    /// Returns the observers listening for this trigger, regardless of target.
```

Comment:
you changed `trigger` to event on a precedent line, but here trigger is back, it's a bit confusing